homeassistant/components/wallbox/switch.py

```diff
@@ -0,0 +1,88 @@
+"""Home Assistant component for accessing the Wallbox Portal API. The lock component creates a switch entity."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WallboxCoordinator, WallboxEntity
+from .const import (
+    CHARGER_DATA_KEY,
+    CHARGER_PAUSE_RESUME_KEY,
+    CHARGER_SERIAL_NUMBER_KEY,
+    CHARGER_STATUS_DESCRIPTION_KEY,
+    DOMAIN,
+)
+
+
+@dataclass
+class WallboxSwitchEntityDescription(SwitchEntityDescription):
+    """Describes Wallbox sensor entity."""
+
+
+SWITCH_TYPES: dict[str, WallboxSwitchEntityDescription] = {
+    CHARGER_PAUSE_RESUME_KEY: WallboxSwitchEntityDescription(
+        key=CHARGER_PAUSE_RESUME_KEY, name="Pause/Resume"
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Create wallbox sensor entities in HASS."""
+    coordinator: WallboxCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [WallboxSwitch(coordinator, entry, SWITCH_TYPES[CHARGER_PAUSE_RESUME_KEY])]
+    )
+
+
+class WallboxSwitch(WallboxEntity, SwitchEntity):
+    """Representation of the Wallbox portal."""
+
+    entity_description: WallboxSwitchEntityDescription
+    coordinator: WallboxCoordinator
+
+    def __init__(
+        self,
+        coordinator: WallboxCoordinator,
+        entry: ConfigEntry,
+        description: WallboxSwitchEntityDescription,
+    ) -> None:
+        """Initialize a Wallbox switch."""
+
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._coordinator = coordinator
+        self._attr_name = f"{entry.title} {description.name}"
+        self._attr_unique_id = f"{description.key}-{coordinator.data[CHARGER_DATA_KEY][CHARGER_SERIAL_NUMBER_KEY]}"
+
+    @property
+    def available(self) -> bool:
+        """Return the availability of the switch."""
+        return self.coordinator.data[CHARGER_STATUS_DESCRIPTION_KEY].lower() in [
+            "charging",
+            "paused",
+            "scheduled",
+        ]
```

Comment:
Two comments here
1. Past two items, `in` comparisons are faster with sets, compared to tuples/list. At least as long as the items are hashable.
2. I saw that you reused the `"charging"` key later. It might be a bit over-engineered, but I would suggest to create a new `StrEnum` (from `homeassistant.backporst.enum`) with all possible `CHARGER_STATUS_DESCRIPTION_KEY` values. Just how the API / library returns them. Then change the check to use the enum values instead. Something like this:
```py
self.coordinator.data[CHARGER_STATUS_DESCRIPTION_KEY] in {
    Status.CHARGING,
    Status.PAUSED,
    Status.SCHEDULED,
}
```

Reply:
Just saw that the name is coming from the `CHARGER_STATUS` mapping in `__init__.py`. In theory, you could replace the strings there with the StrEnum value, too.

Reply:
OK, I think I understand what you mean here. I added a new `StrEnum` class in const and used it in `__init__.py` and `switch.py`